opsdroid/connector/matrix/connector.py

```diff
@@ -31,28 +34,73 @@
 __all__ = ["ConnectorMatrix"]
 
 
+def ensure_room_id_and_send(func):
+    """
+    Ensure that the target for the event is a matrix room id.
+
+    Also retry the function call if the server disconnects.
+    """
```

Comment:
Good call on this. Is there any situation we're not going to want to use this though? Would be good if there were some way to just do this by default on all the functions, without having to add an extra decorator to them.

Reply:
I guess we could override `Connector.send` to do it there, but I think this is probably easier to track.

Reply:
You mean for debugging purposes? Yeah I guess so. But think of all the lines of code you could delete :P

Reply:
We would need the decorator anyway for the conversion to room id, so I really don't think it would save us much.